src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/DocumentProcessor.java

```diff
@@ -0,0 +1,104 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services;
+
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlob;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.function.BiConsumer;
+import java.util.stream.StreamSupport;
+import java.util.zip.ZipFile;
+
+/**
+ * Main component that is run in intervals.
+ *
+ * Looks up all available containers for zip files and extracts pdf files from them.
+ */
+@Component
+public class DocumentProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DocumentProcessor.class);
+
+    private final CloudBlobClient cloudBlobClient;
+    private final BiConsumer<String, ZipFile> handleZip;
```

Comment:
Not that we've got more arguments and everything, maybe it would be better to use a problem-specific interface? By looking at the code you don't know what the string parameter represents.

Also, do we know what the handler will do and need?